packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts

```diff
@@ -937,14 +970,30 @@ const useAzureCommunicationCallAdapterGeneric = <
           adapterRef.current = undefined;
         }
 
-        let newAdapter = await createAzureCommunicationCallAdapter({
-          credential,
-          displayName,
-          locator,
-          userId,
-          /* @conditional-compile-remove(PSTN-calls) */ alternateCallerId,
-          /* @conditional-compile-remove(rooms) */ options
-        });
+        let newAdapter: Adapter;
+        if (adapterKind === 'AzureCommunication') {
+          // This is just the type check to ensure that displayName is defined.
+          if (!('displayName' in args) || !args?.displayName) {
+            throw new Error(
+              'Unreachable code: displayName is not defined, this should have been checked and return earlier.'
+            );
+          }
+          newAdapter = (await createAzureCommunicationCallAdapter({
+            credential,
+            displayName: args.displayName,
+            locator,
+            userId: userId as CommunicationUserIdentifier,
+            /* @conditional-compile-remove(PSTN-calls) */ alternateCallerId: args.alternateCallerId,
+            /* @conditional-compile-remove(rooms) */ options
+          })) as Adapter;
+        } else {
```

Comment:
nit: Since this is specifically a teams call adapter, should we wrap this in an "else if"?
Not really necessary in this case, but it would make it more explicit.

Reply:
Nice catch! Else-if makes lots of sense, then I will add a throw error sentence in last else closure, to make it future proof